Src/HALALMock/Services/DigitalOutputService/DigitalOutputService.cpp

```diff
@@ -34,17 +38,16 @@ void DigitalOutputService::turn_on(uint8_t id){
 	}
 
 	Pin pin = DigitalOutputService::service_ids[id];
-	HAL_GPIO_WritePin(pin.port, pin.gpio_pin, (GPIO_PinState)PinState::ON);
+	(*gpio_memory + pin_offset[pin.gpio_pin]) |= 0x01;
```

Comment:
```suggestion
	(*gpio_memory + pin_offset[pin.gpio_pin]) = PinState::HIGH;
```
don't know the exact name of the enum variant, but you get the idea